packages/ntcore-ts-client/src/lib/pubsub/pubsub.ts

```diff
@@ -53,6 +57,40 @@ export class PubSubClient {
     return instance;
   }
 
+  /**
+   * Adds a listener for when topics are announced.
+   * @param callback - A callback triggered with the announced topic
+   * @param announceAll - Whether to trigger all topics in NetworkTables to be announced
+   * @returns A function to remove the listener.
+   */
+  addOnTopicAnnouncedListener(callback: (_: NetworkTablesTopic) => void, announceAll?: boolean) {
+    this.onTopicAnnouncedListeners.push(callback);
+    if (announceAll) {
+      const subscribeParams: SubscribeMessageParams = {
+        topics: [''],
+        subuid: Util.generateUid(),
+        options: {
+          prefix: true,
+          topicsonly: true,
+        },
+      };
+      this.messenger.subscribe(subscribeParams);
```

Comment:
Based on the below comment, I think we need to (as a library) do this regardless. That way, we always have a source of truth for what topics have been announced.

As I type this, I can't recall if there exists a situation where `onAnnounce` is not called for _every_ topic in existence. Do we need to do some subscription like this as a client to be aware of every topic in existence?

Reply:
From my testing, it seemed like NetworkTables would only announce topics you asked to be notified of via subscriptions (direct or with a prefix), so I do think something like this is needed.

(though maybe I misunderstood your question)

Reply:
That's right, I forgot that the server will notify about an announcement when:

1. The topic has been published already and your new subscription matches it
2. A subscription now has a new published topic